spec/unit/pops/evaluator/evaluating_parser_spec.rb

```diff
@@ -1040,6 +1040,12 @@ def assert_no_undef(x)
       expect{parser.evaluate_string(scope, "assert_no_undef({1 => undef})")}.to_not raise_error()
     end
 
+    it 'a lambda return value is checked using the return type' do
+      expect(parser.evaluate_string(scope, "[1,2].map |Integer $x| >> Integer { $x }")).to eql([1,2])
+      expect { parser.evaluate_string(scope, "[1,2].map |Integer $x| >> String { $x }") }.to raise_error(
+        /value returned from lambda had wrong type, expected a String value, got Integer/)
+    end
+
```

Comment:
The tests serves as examples and specifies what the requirements are. `it "allows any type when return type is not specified"...` It also helps someone reviewing the functionality and trying to figure out if the acceptance criteria are met or not. (Other than that I agree with you - nothing much would work if the requirement is not met).